BookingsApi/BookingsApi/Controllers/PersonsController.cs

```diff
@@ -286,18 +286,18 @@ public async Task<ActionResult<IList<PersonResponse>>> UpdatePersonDetails([From
             var query = new GetHearingsByUsernameQuery(payload.Username);
             var hearings = await _queryHandler.Handle<GetHearingsByUsernameQuery, List<VideoHearing>>(query);
             
-            
             // raise an update event for each hearing to ensure consistency between video and bookings api
-            var createdHearings = hearings.Where(x => x.Status == BookingStatus.Created).ToList();
-            _logger.LogDebug("Updating {Count} confirmed hearing(s)", createdHearings.Count);
-
             var anonymisedText = "@email.net";
-            foreach(var hearing in createdHearings)
+            var nonAnonymisedParticipants = hearings
+                .Where(x => x.Status == BookingStatus.Created &&
+                            x.GetCases().Any(c => !c.Name.EndsWith(anonymisedText))).SelectMany(c => c.Participants)
+                .Where(p => p.PersonId == personId && !p.DisplayName.EndsWith(anonymisedText)).ToList();
```

Comment:
Possible NULL display name ?

Reply:
Oh that is a very good point. I couldn't see null's in the DB but it's not mandatory. I will look for another property.